doc/cascadia/Non-Terminal-Panes.md

```diff
@@ -0,0 +1,115 @@
+
+When we want to host other types of content, we'll simply pass any other control
+to the Pane, and it'll render it just as it would the `TermControl`.
+
+## UI/UX Design
+
+Instead of a pane hosting a terminal, it could host _any arbitrary control_. The
+control would still be subject to the sizing provided to it by the `Pane`, but
+it could host any arbitrary content.
+
+## Capabilities
+
+### Security
+
+I don't forsee this implementation by itself raising security concerns. This
+feature is mostly concerned with adding support for arbitrary controls, not
+actually implementing some arbitrary controls.
+
+### Reliability
+
+With more possible controls in a pane than just a terminal, it's possible that
+crashes in those controls could impact the entire Terminal app's reliability.
+This would largely be out of our control, as we only author the TermControl.
+
+We may want to consider hosting each pane in it's own process, similar to how
+moder browsers will host each tab in its own process, to help isolate tabs. This
+is a bigger discussion than the feature at hand, however.
+
+### Performance, Power, and Efficiency
+
+I don't believe this will have a noticable impact _on its own_. Should the user
+decide to host a WebView in a pane, then it surely could impact these measuress.
+However, I leave that discussion to the implementation of the actual alternative
+pane content itself.
+
+### Accessibility
+
+When implementing the accessibility tree for Panes, we'll need to make sure that
+for panes with arbitrary content, we properly activate their accessibility,
+should the control provide some sort of accessibility pattern.
+
+## Potential Issues
+
+* [ ] It's entirely possible that panes with non-terminal content will not be
+  able to activate keybindings from the terminal application. For example, what
+  if the hosted control wants to use Ctrl+T for its own shortcut? The current
+  keybindings model has the `TermControl` call into the App layer to see if a
+  keystroke should be handled by the app first. We may want to make sure that
+  for non-terminal controls, we add a event handler to try and have the
+  `AppKeyBindings` handle the keypress ifthe control doesn't. This won't solve
+  the case where the control wants to use a keybinding that is mapped by the
+  Terminal App. In that case, non-terminal controls will actually behave
+  differently from the `TermControl`. The `TermControl` will give the app the
+  first chance to handle a keybinding, while for other controls, the app will
+  give the control the first chance to handle the keypress. This may be mildly
+  confusing to end users.
+
+## Future considerations
+
+I expect this to be a major part of our (eventual) extensibility model. By
+allowing arbitrary controls to be hosted in a Tab/Pane, this will allow
+extension authors to embed their own UI experiences alongside the terminal.
+See #555 for more discussion on the extensibility/plugin subject.
+
+## Resources
+
+N/A

```

Comment:
@zadjii-msft almost embarrassed to make the nit comments.   I am not sure the longevity of these specs and the nits probably can be ignored. This is a good spec.   The keybinding issue is a good one.  This is always tricky when hosting random controls.    Hosting an external process in a tab is definitely a bigger discussion.  Nice writeup.

Reply:
Lol don't worry about it - I type up specs in a text editor, without spellcheck, so the nits are really helpful, cause I obviously didn't proofread it myself :P